lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -935,16 +935,38 @@ static Value tryEliminatingConnectsToValue(Value flipValue,
                 connectSrc)
             .getResult(0);
 
-  // We know it must be the destination operand due to the types, but the
-  // source may not match the destination width.
-  auto destTy = flipValue.getType().cast<FIRRTLType>().getPassiveType();
-  if (destTy.getBitWidthOrSentinel() !=
-      connectSrc.getType().cast<FIRRTLType>().getBitWidthOrSentinel()) {
-    // The only type mismatchs we care about is due to integer width
-    // differences.
-    auto destWidth = destTy.getBitWidthOrSentinel();
-    assert(destWidth != -1 && "must know integer widths");
-    connectSrc = builder.createOrFold<PadPrimOp>(destTy, connectSrc, destWidth);
+  if (connectSrc.getType().isa<FVectorType>()) {
```

Comment:
I would recommend dropping all this for now: tryEliminatingConnectsToValue is just an optimization.  As you're working towards correctness you don't need it.  Just conditionalize this logic (that inserts the pad) on whether the src/dest are firrtl integer types.  If not, and if they don't match, just bail out conservatively.